core/domain/exp_domain.py

```diff
@@ -1224,6 +1224,46 @@ def create_default_interaction(cls, default_dest_state_name):
             cls._DEFAULT_INTERACTION_ID, {}, [], default_outcome, [], [], {})
 
 
+    def get_all_html_content_strings(self):
+        """Get all html content strings in the interaction.
+
+        Returns:
+            list(str): The list of all html content strings in the interaction.
+        """
+        html_list = []
+
+        for answer_group in self.answer_groups:
+            outcome_html = answer_group.outcome.feedback.html
+            html_list = html_list + [outcome_html]
+
+        if self.id == 'ItemSelectionInput':
+            for answer_group in self.answer_groups:
+                for rule_spec in answer_group.rule_specs:
+                    rule_spec_html = rule_spec.inputs['x']
+                    html_list = html_list + rule_spec_html
+
+        if self.default_outcome:
+            default_outcome_html = self.default_outcome.feedback.html
+            html_list = html_list + [default_outcome_html]
+
+
+        for hint in self.hints:
+            hint_html = hint.hint_content.html
+            html_list = html_list + [hint_html]
+
+
+        if self.solution:
+            solution_html = self.solution.explanation.html
+            html_list = html_list + [solution_html]
+
+        if self.id in ('ItemSelectionInput', 'MultipleChoiceInput'):
+            customization_args_html = (
```

Comment:
Here and elsewhere, differentiate between html and html list.

Reply:
Done!